plugins/search/providers/elastic_search/lib/model/filters/ESearchQueryFromAdvancedSearch.php

```diff
@@ -129,29 +129,87 @@ protected function createESearchQueryFromSearchFilterOperator(AdvancedSearchFilt
+
+		if ($searchFilter->getHasCaption())
+		{
+			$hasCaptionsItem = new ESearchCaptionItem();
+			$hasCaptionsItem->setFieldName(ESearchCaptionFieldName::CONTENT);
+			$hasCaptionsItem->setItemType(ESearchItemType::EXISTS);
+			$captionFieldConditions[] = $hasCaptionsItem;
+		}
+
+		if ($searchFilter->getLanguage())
 		{
-			$item->setFieldName(ESearchCaptionFieldName::CONTENT);
-			$item->setItemType(ESearchItemType::EXISTS);
+			$language = $searchFilter->getLanguage();
+			$languageItem = new ESearchCaptionItem();
+			$languageItem->setFieldName(ESearchCaptionFieldName::LANGUAGE);
+			$languageItem->setItemType(ESearchItemType::EXACT_MATCH);
+			$languageItem->setSearchTerm($language);
+			$captionFieldConditions[] = $languageItem;
+		}
+
+		if ($searchFilter->getAccuracyGreaterThanOrEqual() !== null || $searchFilter->getAccuracyLessThanOrEqual() !== null ||
+			$searchFilter->getAccuracyGreaterThan() !== null || $searchFilter->getAccuracyLessThan() !== null)
+		{
+			$accuracyRange = new ESearchRange();
+			if ($searchFilter->getAccuracyGreaterThanOrEqual() !== null)
+			{
+				$accuracyRange->setGreaterThanOrEqual($searchFilter->getAccuracyGreaterThanOrEqual());
+			}
+			if ($searchFilter->getAccuracyLessThanOrEqual() !== null)
+			{
+				$accuracyRange->setLessThanOrEqual($searchFilter->getAccuracyLessThanOrEqual());
+			}
+			if ($searchFilter->getAccuracyGreaterThan() !== null)
+			{
+				$accuracyRange->setGreaterThan($searchFilter->getAccuracyGreaterThan());
+			}
+			if ($searchFilter->getAccuracyLessThan() !== null)
+			{
+				$accuracyRange->setLessThan($searchFilter->getAccuracyLessThan());
+			}
+
+			$accuracyItem = new ESearchCaptionItem();
+			$accuracyItem->setRange($accuracyRange);
+			$accuracyItem->setFieldName(ESearchCaptionFieldName::ACCURACY);
+			$accuracyItem->setItemType(ESearchItemType::RANGE);
+			$captionFieldConditions[] = $accuracyItem;
+		}
+
+		if ($searchFilter->getUsage() !== null)
+		{
+			$usage = $searchFilter->getUsage();
+			$usageItem = new ESearchCaptionItem();
+			$usageItem->setFieldName(ESearchCaptionFieldName::USAGE);
+			$usageItem->setItemType(ESearchItemType::EXACT_MATCH);
+			$usageItem->setSearchTerm($usage);
+			$captionFieldConditions[] = $usageItem;
+		}
+
+		if (empty($captionFieldConditions))
+		{
+			return null;
+		}
+
+		// Create the nested query ensuring all conditions apply to the same caption asset
+		if (count($captionFieldConditions) === 1)
+		{
+			$nestedQuery = $captionFieldConditions[0];
 		}
 		else
 		{
-			$item->setFieldName(ESearchCaptionFieldName::LANGUAGE);
-			$item->setItemType(ESearchItemType::EXACT_MATCH);
-			$item->setSearchTerm($searchFilter->getLanguage());
+			$nestedQuery = self::createNestedOperator($captionFieldConditions);
 		}
-		$result = $item;
 
 		if ($searchFilter->getHasCaption())
 		{
-			return $result;
+			return $nestedQuery;
 		}
 		else
 		{
-			return self::createNegativeQuery($item);
+			return self::createNegativeQuery($nestedQuery);
```

Comment:
the legacy was has captions or does not have captions => this is why we applied the "not"
details:  
https://github.com/kaltura/server/blame/7008342888b58f13976df54fe2d9259265fe9fc7/plugins/search/providers/elastic_search/lib/model/filters/ESearchQueryFromAdvancedSearch.php#L115C47-L115C47
- create query of exists 
- if $searchFilter->getHasCaption() == true ==> return the query created 
- if not $searchFilter->getHasCaption() ==> return "not" the query created 



now, with this code, by this point 
the entire query analyzed above will be under "not" ( in case of no $searchFilter->getHasCaption() )

e.g. 
if the query was only about accuracy greater than 50 
since we did not pass  $searchFilter->getHasCaption(), it will create query with accuracy not greater than 50